### PR TITLE
[split/7] move built-in scalar function to datafusion-expr

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

- part of #1750 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion-expr/src/lib.rs [hunk 1]
```diff
@@ -16,12 +16,14 @@
 // under the License.
 
 mod aggregate_function;
```
Comment (alamb) [inline datafusion-expr/src/lib.rs:18]: I also like how this reorganization is making the structure of `Expr` easier to understand

File: datafusion-expr/src/built_in_function.rs [hunk 2]
```diff
@@ -0,0 +1,330 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (alamb) [inline datafusion-expr/src/built_in_function.rs:1]: is the plan to move the built in aggregate functions here too?

**Review by alamb — APPROVED** [review]

File: datafusion-expr/src/built_in_function.rs (hunk 2 above)
Comment (Jimexist) [reply datafusion-expr/src/built_in_function.rs:1]: i think it's in the `aggregate_function` module

File: datafusion-expr/src/lib.rs (hunk 1 above)
Comment (Jimexist) [reply datafusion-expr/src/lib.rs:18]: these:
- #1774 
- #1784 

is also trying to tear up the cumbersome/bloated expr by traits.

Comment (Jimexist) [reply datafusion-expr/src/lib.rs:18]: so that each trait module can be migrated separately

Comment (alamb) [reply datafusion-expr/src/lib.rs:18]: > is also trying to tear up the cumbersome/bloated expr by traits.

🏅

**Pull request merged**